app/api/private_ai_keys.py

```diff
@@ -390,6 +412,67 @@ async def list_private_ai_keys(
     private_ai_keys = query.all()
     return [key.to_dict() for key in private_ai_keys]
 
+@router.get("/{key_id}", response_model=PrivateAIKeyDetail, dependencies=[Depends(check_system_admin)])
+async def get_private_ai_key(
+    key_id: int,
+    current_user = Depends(get_current_user_from_auth),
+    db: Session = Depends(get_db)
+):
+    """
+    Get details of a specific private AI key.
+
+    This endpoint will:
+    1. Verify the user has access to the key
+    2. Return the full details of the key including LiteLLM-specific data
+
+    Required parameters:
+    - **key_id**: The ID of the private AI key to retrieve
+
+    The response will include:
+    - Database connection details (host, database name, username, password)
+    - LiteLLM API token for authentication
+    - LiteLLM API URL for making requests
+    - Owner and team information
+    - Region information
+    - LiteLLM-specific data (spend, duration, budget, etc.)
+
+    Note: You must be authenticated to use this endpoint.
+    Only system administrators can access this endpoint.
+    """
+    private_ai_key = _get_key_if_allowed(key_id, current_user, "system_admin", db)
+
+    # Get the region
+    region = db.query(DBRegion).filter(DBRegion.id == private_ai_key.region_id).first()
+    if not region:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Region not found"
+        )
+
+    # Create LiteLLM service instance
+    litellm_service = LiteLLMService(
+        api_url=region.litellm_api_url,
+        api_key=region.litellm_api_key
+    )
+
+    try:
+        # Get LiteLLM key info
+        litellm_data = await litellm_service.get_key_info(private_ai_key.litellm_token)
+        info = litellm_data.get("info", {})
+        logger.info(f"LiteLLM key info: {info}")
+
+        # Combine database key info with LiteLLM info
+        key_data = private_ai_key.to_dict()
+        key_data.update(info)
```

Comment:
I was looking at [this commit](https://github.com/amazeeio/amazee.ai/pull/56/commits/3ba1d8e09164163f412209cd49670f3239658bde#diff-5de65c6fdfae351c3d108d7689ccd69c3701f6b8ce3a4f9e2ff5f9d7ce3e3258R466) and noticed that it previously changed "expires" to "expires_at". just flagging in case that's a bug/problem.